x-pack/plugins/apm/public/components/app/ErrorGroupDetails/Distribution/index.tsx

```diff
@@ -54,76 +43,72 @@ export function getFormattedBuckets(
+  const xMax = d3.max(buckets, (d) => d.x0);
+
+  const xFormatter = niceTimeFormatter([xMin, xMax]);
+
+  const tooltipProps: SettingsSpec['tooltip'] = {
+    headerFormatter: (tooltip: TooltipValue) => {
+      const serie = buckets.find((bucket) => bucket.x0 === tooltip.value);
+      if (serie) {
+        return asRelativeDateTimeRange(serie.x0, serie.x);
+      }
+      return `${tooltip.value}`;
+    },
+  };
 
   return (
     <div>
       <EuiTitle size="xs">
         <span>{title}</span>
       </EuiTitle>
-      <Histogram
-        height={180}
-        noHits={distribution.noHits}
-        tooltipHeader={tooltipHeader}
-        verticalLineHover={(bucket: FormattedBucket) => bucket.x}
-        xType="time-utc"
-        formatX={(value: Date) => {
-          const time = value.getTime();
-          return tickFormat(new Date(time - getTimezoneOffsetInMs(time)));
-        }}
-        buckets={buckets}
-        bucketSize={distribution.bucketSize}
-        formatYShort={(value: number) =>
-          i18n.translate('xpack.apm.errorGroupDetails.occurrencesShortLabel', {
-            defaultMessage: '{occCount} occ.',
-            values: { occCount: value },
-          })
-        }
-        formatYLong={(value: number) =>
-          i18n.translate('xpack.apm.errorGroupDetails.occurrencesLongLabel', {
-            defaultMessage:
-              '{occCount} {occCount, plural, one {occurrence} other {occurrences}}',
-            values: { occCount: value },
-          })
-        }
-        legends={[
-          {
-            color: theme.euiColorVis1,
-            // 0a abbreviates large whole numbers with metric prefixes like: 1000 = 1k, 32000 = 32k, 1000000 = 1m
-            legendValue: numeral(averageValue).format('0a'),
-            title: i18n.translate('xpack.apm.errorGroupDetails.avgLabel', {
-              defaultMessage: 'Avg.',
-            }),
-            legendClickDisabled: true,
-          },
-        ]}
-      />
+      <div style={{ height: 180 }}>
+        <Chart>
+          <Settings
+            xDomain={{ min: xMin, max: xMax }}
+            tooltip={tooltipProps}
+            showLegend
+            showLegendExtra
+            legendPosition={Position.Bottom}
+          />
+          <Axis
+            id="x-axis"
+            position={Position.Bottom}
+            showOverlappingTicks
+            tickFormat={xFormatter}
+          />
+          <Axis
+            id="y-axis"
+            position={Position.Left}
+            ticks={2}
+            showGridLines
+            tickFormat={(value) => `${value} occ.`}
+          />
+          <HistogramBarSeries
+            minBarHeight={2}
+            id="errorOccurrences"
+            name="Occurences"
+            xScaleType={ScaleType.Linear}
+            yScaleType={ScaleType.Linear}
+            xAccessor="x0"
+            yAccessors={['y']}
+            data={distribution.noHits ? [] : buckets}
```

Comment:
I think we should get rid of `noHits` and never render the chart if there's no data. Instead we should render placeholders, something like:

```ts

if(!data && status === FETCH_STATUS.LOADING) {
  return <ChartLoadingPlaceholder />
}

if(!data && status === FETCH_STATUS.SUCCESS) {
  return <ChartEmptyPlaceholder />
}

return <Chart />
```

Reply:
I think `Loading` and `Failed` would be nice to have, but `Empty` Elastic charts has an OOTB message when there's no data, maybe we shouldn't overwrite it. WDYT?

Reply:
Okay, that's fine with me - but do we still need the `noHits`? Would this work?
```suggestion
            data={buckets}
```

Reply:
> but do we still need the noHits? Would this work?

I changed the return from the API to return an empty array if there are no hits, currently, it returns an array of timeseries with value of 0. Now the `distribution.noHits` is no longer needed.

Reply:
> currently, it returns an array of timeseries with value of 0.

Yeah, that was a workaround for displaying the empty state. We should move away definitely get rid of this.

Reply:
I still think we should have explicit loading and error states (unless that's also part of the elastic charts)

Reply:
It is not, but the `ChartContainer` handles that. https://github.com/cauemarcondes/kibana/blob/apm-elastic-charts/x-pack/plugins/apm/public/components/shared/charts/chart_container.tsx#L20-L32